Code/Tools/FBuild/FBuildCore/BFF/Functions/FunctionObjectList.cpp

```diff
@@ -212,6 +212,8 @@ FunctionObjectList::FunctionObjectList()
 		return false;
 	}
 
+	const char * baseDirectory = GetBaseDirectory( funcStartIter );
```

Comment:
There are a few problems here:
**a) Not "cleaning" the path.** This will work currently only if you provide exactly the right full path. Cleaning the path takes care of canonicalizing the path.
**b) Not checking for errors.** If the bff has a syntax error, we need to return instead of continuing parsing.
**c) .CompilerInputFiles is not optional.** We want this to be opt-in, since it's only when listing files explicitly that this is needed.
We need something like this:
```C++
AStackString<> baseDir, baseDirClean;
if ( !GetString( funcStartIter, baseDir, ".CompilerInputFilesRoot", false ) ) // false = optional
{
    return false; // GetString will have emitted error
}
if ( !baseDir.IsEmpty() )
{
    NodeGraph::CleanPath( baseDir, baseDirClean );
}
```
Then you can always pass "baseDirClean" through to CreateLibraryNode.